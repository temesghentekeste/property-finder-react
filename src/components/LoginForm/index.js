import { useState, useRef, useEffect } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import HomeIcon from '@material-ui/icons/Home';
import common from '../../common/common.module.css';
import styles from './LoginForm.module.css';

const LoginForm = ({ handleSubmit }) => {
  const [user, setUser] = useState({
    username: '',
    password: '',
  });

  const userNameRef = useRef();
  useEffect(() => {
    userNameRef.current.focus();
  }, []);

  const handleChange = (e) => {
    const { name, value } = e.target;

    setUser({ ...user, [name]: value });
  };

  return (
    <div className={`${styles.formContainer} ${styles.content}`} data-testid="loginForm">
      <header className={styles.formContainer_header}>
        <h2 data-testid="loginForm-heading">Sign In</h2>
        <p>Hello there! Log in and start managing your system.</p>
      </header>

      <form onSubmit={(e) => handleSubmit(e, user)}>
        <div className={styles.formContainer__formInput}>
          <input
            type="text"
            id="username"
            name="username"
            value={user.username}
            onChange={handleChange}
            ref={userNameRef}
          />
        </div>
        <div className={styles.formContainer__formInput}>
          <input
            type="password"
            id="password"
            name="password"
            value={user.password}
            onChange={handleChange}
          />
        </div>
        <div className={styles.formContainer__formInput}>
          <button type="submit" className={styles.formContainer_btn}>Sign In</button>
        </div>
      </form>

      <div className={styles.signup}>
        <span>If you are new, please,&nbsp;</span>
        <Link to="/signup">Sign up</Link>
      </div>

      <div className={common.backToHome}>
        <Link to="/">
          Back to&nbsp;
          <HomeIcon />
        </Link>
      </div>
    </div>
  );
};

LoginForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
};

export default LoginForm;
