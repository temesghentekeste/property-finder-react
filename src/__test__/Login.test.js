/* eslint-disable no-console */

import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import LoginForm from '../components/LoginForm';

let getByTestId;
const handleSubmit = () => console.log('Handled');
beforeEach(() => {
  const component = render(
    <Router>
      <LoginForm handleSubmit={handleSubmit} />
    </Router>,
  );
  getByTestId = component.getByTestId;
});

it('renders the LoginForm component', () => {
  const component = renderer
    .create(
      <Router>
        <LoginForm handleSubmit={handleSubmit} />
      </Router>,
    )
    .toJSON();

  expect(component).toMatchSnapshot();
});

test('should render LoginForm component with correct heading text', () => {
  const headingText = getByTestId('loginForm-heading');

  expect(headingText).toHaveTextContent('Sign In');
});
