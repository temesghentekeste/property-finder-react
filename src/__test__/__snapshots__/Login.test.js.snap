// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the LoginForm component 1`] = `
<div
  className="formContainer content"
  data-testid="loginForm"
>
  <header
    className="formContainer_header"
  >
    <h2
      data-testid="loginForm-heading"
    >
      Sign In
    </h2>
    <p>
      Hello there! Log in and start managing your system.
    </p>
  </header>
  <form
    onSubmit={[Function]}
  >
    <div
      className="formContainer__formInput"
    >
      <input
        id="username"
        name="username"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="formContainer__formInput"
    >
      <input
        id="password"
        name="password"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div
      className="formContainer__formInput"
    >
      <button
        className="formContainer_btn"
        type="submit"
      >
        Sign In
      </button>
    </div>
  </form>
  <div
    className="signup"
  >
    <span>
      If you are new, please, 
    </span>
    <a
      href="/signup"
      onClick={[Function]}
    >
      Sign up
    </a>
  </div>
  <div
    className="backToHome"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Back to 
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
        />
      </svg>
    </a>
  </div>
</div>
`;
